inshatan/nodejs-aws-be
# task-9

# Task 9 (Backend For Frontend)


#### product-service

- endpoint: https://ojlkuvsx07.execute-api.eu-west-1.amazonaws.com/dev
- all products: https://ojlkuvsx07.execute-api.eu-west-1.amazonaws.com/dev/products/ 
- single product: https://ojlkuvsx07.execute-api.eu-west-1.amazonaws.com/dev/product/030f5ee8-b461-4543-ac4b-103032945249


#### cart-service

- endpoint: http://inshatan-cart-api-dev.eu-west-1.elasticbeanstalk.com
- user cart:  http://inshatan-cart-api-dev.eu-west-1.elasticbeanstalk.com/api/profile/cart

#### *bff-service*

- endpoint: http://inshatan-bff-api-dev.eu-west-1.elasticbeanstalk.com
- all products: http://inshatan-bff-api-dev.eu-west-1.elasticbeanstalk.com/products/products
- single product: http://inshatan-bff-api-dev.eu-west-1.elasticbeanstalk.com/products/product/030f5ee8-b461-4543-ac4b-103032945249
- user cart:  http://inshatan-bff-api-dev.eu-west-1.elasticbeanstalk.com/cart/api/profile/cart

#### Create product

To create product you can use Frontend Application (https://d2v14ttltat414.cloudfront.net).
   
Please use a csv file in the form (example can be found in the file **./products.csv**): 
  
    title,description,price,count
    
 
## Main tasks

 
 
## Additional (optional) tasks

How to test:
  * Get products list
  * Create new product
  * Get products list - result shouldn’t have new product
  * Wait more than 2 minutes
  * Get products list - result should have new product
  **!!! service with NestJS is located in the ./bff-service-nest**
